source/_posts/2017-11-02-kafka-streams-dsl-vs-processor-api.markdown

```diff
@@ -0,0 +1,542 @@
+---
+layout: post
+title: "Kafka Streams DSL vs Processor API"
+date: 2017-11-02
+comments: true
+categories: [kafka, kafka streams, scala]
+---
+
+[Kafka Streams](https://docs.confluent.io/current/streams/index.html) is a Java library 
+for building real-time, highly scalable, fault tolerant, distributed applications. 
+The library is fully integrated with [Kafka](https://kafka.apache.org/documentation/) and leverages 
+Kafka producer and consumer semantics (e.g: partitioning, rebalancing, data retention and compaction).
+What is really unique, a running Kafka cluster is the only dependency required to run Kafka Stream application.
+Even local state stores are backed by Kafka topics to make the processing fault tolerant - brilliant!
+
+Kafka Streams provides all necessary stream processing primitives like one-record-at-a-time processing, 
+event time processing, windowing support and local state management. 
+Application developer can choose from three different Kafka Streams APIs: DSL, Processor or KSQL.
+
+* [Kafka Streams DSL](https://docs.confluent.io/current/streams/developer-guide.html#kafka-streams-dsl) 
+(Domain Specific Language) - recommended way for most users 
+because business logic can be expressed in a few lines of code.
+All stateless and stateful transformations are defined using declarative, 
+functional programming style (filter, map, flatMap, reduce, aggregate operations).
+Kafka Stream DSL encapsulates most of the stream processing complexity
+but unfortunately it also hides many useful knobs and switches. 
+
+* [Kafka Processor API](https://docs.confluent.io/current/streams/developer-guide.html#processor-api) 
+provides a low level, imperative way to define stream processing logic.
+At first sight Processor API could look hostile but finally gives much more flexibility to developer.
+This blog post shows that hand-crafted stream processors might be a magnitude more efficient than
```

Comment:
zamieniłbym to chyba na:
This blog post demonstrates
lub nawet lepiej na:
With this blog post I would like to demonstrate